solr/core/src/java/org/apache/solr/handler/loader/JsonLoader.java

```diff
@@ -703,6 +703,10 @@ private SolrInputDocument generateExtendedValueDoc(int ev) throws IOException {
         String label = parser.getString();
         SolrInputField sif = new SolrInputField(label);
         parseFieldValue(sif);
+        // pulling out the pieces may seem weird, but it's because
```

Comment:
Good.  The duplication of comment now has me wondering about this method overall -- why does it seem to be duplicating existing logic -- buildDoc?  Would things "just work" if we replaced generateExtendedValueDoc with buildDoc()?  I think it would?  Less look-a-like code to maintain.  I suppose the only problem with that is it would permit weird things like anonymous child docs on a labelled child doc.  I just wouldn't worry about that now though; we catch that later in AddUpdateCommand.flatten().

BTW I'm sure you realize now that I don't have everything figured out from the start; it's an iterative journey for you and me/the-reviewer.  :-)

Reply:
It's actually quite refreshing to get another person's opinion.
I don't think it would work flawlessly without changing a lot of the code, since buildDoc expects a Map as an input, so we would have to parse the map beforehand.

Reply:
Skimming through the code on master, it looks as if the map parsing was done here before only partially. Currently it seems like we either duplicate the map parsing part, or the document parsing part. If we go with the map part, we would build an intermediate map and then parse it to a document instead of streaming it using the parser. I'm not quite sure if using buildDoc would be as beneficial, since in either case, we have to duplicate some part.

Reply:
Sorry I meant `parseDoc`

Reply:
True, didn't quite notice it, I'll upload a commit soon. :smile:

Reply:
It worked.